src/cascadia/TerminalSettingsModel/AIConfig.cpp

```diff
@@ -0,0 +1,205 @@
+        PasswordCredential cred;
+        try
+        {
+            cred = vault.Retrieve(PasswordVaultResourceName, PasswordVaultAIEndpoint);
+        }
+        catch (...)
+        {
+            // there was nothing to remove, just return
+            return;
+        }
+        vault.Remove(cred);
+    }
+    else
+    {
+        PasswordCredential newCredential{ PasswordVaultResourceName, PasswordVaultAIEndpoint, endpoint };
+        vault.Add(newCredential);
+    }
+}
+
+winrt::hstring AIConfig::AzureOpenAIKey() noexcept
+{
+    PasswordVault vault;
+    PasswordCredential cred;
+    // Retrieve throws an exception if there are no credentials stored under the given resource so we wrap it in a try-catch block
+    try
+    {
+        cred = vault.Retrieve(PasswordVaultResourceName, PasswordVaultAIKey);
+    }
+    catch (...)
+    {
+        return L"";
+    }
+    return cred.Password();
+}
+
+void AIConfig::AzureOpenAIKey(const winrt::hstring& key) noexcept
+{
+    PasswordVault vault;
+    if (key.empty())
+    {
+        // the user has entered an empty string, that indicates that we should clear the key
+        PasswordCredential cred;
+        try
+        {
+            cred = vault.Retrieve(PasswordVaultResourceName, PasswordVaultAIKey);
+        }
+        catch (...)
+        {
+            // there was nothing to remove, just return
+            return;
+        }
+        vault.Remove(cred);
+    }
+    else
+    {
+        PasswordCredential newCredential{ PasswordVaultResourceName, PasswordVaultAIKey, key };
+        vault.Add(newCredential);
+    }
+}
+
+winrt::hstring AIConfig::OpenAIKey() noexcept
+{
+    PasswordVault vault;
+    PasswordCredential cred;
+    // Retrieve throws an exception if there are no credentials stored under the given resource so we wrap it in a try-catch block
+    try
+    {
+        cred = vault.Retrieve(PasswordVaultResourceName, PasswordVaultOpenAIKey);
+    }
+    catch (...)
+    {
+        return L"";
+    }
+    return cred.Password();
+}
+
+void AIConfig::OpenAIKey(const winrt::hstring& key) noexcept
+{
+    PasswordVault vault;
+    if (key.empty())
+    {
+        // the user has entered an empty string, that indicates that we should clear the key
+        PasswordCredential cred;
+        try
+        {
+            cred = vault.Retrieve(PasswordVaultResourceName, PasswordVaultOpenAIKey);
+        }
+        catch (...)
+        {
+            // there was nothing to remove, just return
+            return;
+        }
+        vault.Remove(cred);
+    }
+    else
+    {
+        PasswordCredential newCredential{ PasswordVaultResourceName, PasswordVaultOpenAIKey, key };
+        vault.Add(newCredential);
+    }
+}
+
+winrt::Microsoft::Terminal::Settings::Model::LLMProvider AIConfig::ActiveProvider()
+{
+    const auto val{ _getActiveProviderImpl() };
+    if (val)
+    {
+        // an active provider was explicitly set, return that
+        return *val;
+    }

```

Comment:
This does mean that you can set an active provider even if you have not configured that provider - in this case when you try to use the Chat the provider will return the appropriate error. For example, the OpenAI provider will still make the http request and we get an error message back that no api key was provided.